msm/src/serialization/witness.rs

```diff
@@ -274,17 +278,49 @@ pub fn deserialize_field_element<Fp: Field, Env: InterpreterEnv<Fp>>(
 
     env.add_constraint(constraint);
 
+    // -- Start third constraint
     // FIXME: range check
-    let c12 = Env::get_column_for_msm_limb(12);
-    let c13 = Env::get_column_for_msm_limb(13);
-    let c14 = Env::get_column_for_msm_limb(14);
-    let c15 = Env::get_column_for_msm_limb(15);
-    let c16 = Env::get_column_for_msm_limb(16);
-    env.bitmask_be(&input_limb2, 19, 4, c12);
-    env.bitmask_be(&input_limb2, 34, 19, c13);
-    env.bitmask_be(&input_limb2, 49, 34, c14);
-    env.bitmask_be(&input_limb2, 64, 49, c15);
-    env.bitmask_be(&input_limb2, 79, 64, c16);
+    let c12_var = {
+        let c12 = Env::get_column_for_msm_limb(12);
+        env.bitmask_be(&input_limb2, 19, 4, c12)
+    };
+    let c13_var = {
+        let c13 = Env::get_column_for_msm_limb(13);
+        env.bitmask_be(&input_limb2, 34, 19, c13)
+    };
+
+    let c14_var = {
+        let c14 = Env::get_column_for_msm_limb(14);
+        env.bitmask_be(&input_limb2, 49, 34, c14)
+    };
+
+    let c15_var = {
+        let c15 = Env::get_column_for_msm_limb(15);
+        env.bitmask_be(&input_limb2, 64, 49, c15)
+    };
+    let c16_var = {
+        let c16 = Env::get_column_for_msm_limb(16);
+        env.bitmask_be(&input_limb2, 79, 64, c16)
+    };
+
+    // Unfolding for readability.
```

Comment:
makes it less readable to me 😅

Reply:
Ok, let me change it then.

Reply:
Lemme know if it is better for you: https://github.com/o1-labs/proof-systems/pull/1901